utility/src/main/java/org/oppia/android/util/logging/firebase/FirebaseEventLogger.kt

```diff
@@ -18,20 +19,33 @@ private const val COUNTRY_USER_PROPERTY = "COUNTRY"
 class FirebaseEventLogger(
   private val firebaseAnalytics: FirebaseAnalytics,
   private val eventBundleCreator: EventBundleCreator,
-  context: Context
+  private val networkConnectionUtil: NetworkConnectionUtil
 ) : EventLogger {
   private var bundle = Bundle()
-  private val connectivityManager: ConnectivityManager =
-    context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+
+  @Inject
+  lateinit var context: Context
 
   /** Logs an event to Firebase Analytics with [NETWORK_USER_PROPERTY] and [COUNTRY_USER_PROPERTY]. */
   override fun logEvent(eventLog: EventLog) {
     bundle = eventBundleCreator.createEventBundle(eventLog)
     firebaseAnalytics.logEvent(eventLog.actionName.toString(), bundle)
     // TODO(#3792): Remove this usage of Locale.
     firebaseAnalytics.setUserProperty(COUNTRY_USER_PROPERTY, Locale.getDefault().displayCountry)
-    firebaseAnalytics.setUserProperty(
-      NETWORK_USER_PROPERTY, connectivityManager.activeNetworkInfo.typeName
-    )
+    firebaseAnalytics.setUserProperty(NETWORK_USER_PROPERTY, getNetworkStatus())
+  }
+
+  private fun getNetworkStatus(): String {
+    return when (networkConnectionUtil.getCurrentConnectionStatus()) {
+      NetworkConnectionUtil.ProdConnectionStatus.LOCAL ->
+        context.resources.getString(NetworkConnectionUtil.ProdConnectionStatus.LOCAL.connectionName)
+      NetworkConnectionUtil.ProdConnectionStatus.CELLULAR ->
+        context.resources.getString(
+          NetworkConnectionUtil.ProdConnectionStatus.CELLULAR.connectionName
+        )
+      else -> context.resources.getString(
+        NetworkConnectionUtil.ProdConnectionStatus.NONE.connectionName
```

Comment:
@BenHenning And extracting the string in this way, 
Most of the places we use `AppLanguageResourceHandler` to extract string from resources. Let me know if this approach is GTG or we need something else here. 

Injected the context on field, will update it to constructor one.